### PR TITLE
feat: add logging of Monero PoW data to debug merge mining

### PR DESCRIPTION
**Pull request by philipr-za**
Description
---
An intermittent error is being observed where Xmrig submits what it considers a valid PoW solution but the Base Node validation rejects the block with a low (1, 2, 3) difficulty.

Adding two sets of logs to provide more detail on the data used to perform the RandomX hash when verifying a submitting merge mined block. Added a log to the  Merge Mining Proxy to show the full Monero PoW data received from Xmrig and added a log in the Base Node to show the RandomX hash that it produces.

Motivation and Context
---
These logs will help us confirm where the issue is in this disconnect between Xmrig’s calculation and our validation.

How Has This Been Tested?
---
Manually

### CODE REVIEW

**Review by delta1 — APPROVED** [review]
LGTM

File: applications/tari_merge_mining_proxy/src/proxy.rs [hunk 1]
```diff
@@ -276,6 +276,8 @@ impl InnerService {
 
             let monero_data = monero_rx::construct_monero_data(monero_block, block_data.monero_seed.clone())?;
 
+            debug!(target: LOG_TARGET, "Monero PoW Data: {:?}", monero_data);
```
Comment (stringhandler) [inline applications/tari_merge_mining_proxy/src/proxy.rs:279]:
```suggestion
            debug!(target: LOG_TARGET, "Monero PoW Data: {:?}, seed:{}", monero_data, block_data.monero_seed);
```

File: base_layer/core/src/proof_of_work/monero_rx/helpers.rs [hunk 2]
```diff
@@ -51,6 +51,7 @@ pub fn monero_difficulty(header: &BlockHeader, randomx_factory: &RandomXFactory)
 
 fn get_random_x_difficulty(input: &[u8], vm: &RandomXVMInstance) -> Result<(Difficulty, Vec<u8>), MergeMineError> {
     let hash = vm.calculate_hash(&input)?;
+    debug!(target: LOG_TARGET, "RandomX Hash: {:?}", hash);
```
Comment (stringhandler) [inline base_layer/core/src/proof_of_work/monero_rx/helpers.rs:54]:
```suggestion
    debug!(target: LOG_TARGET, "RandomX Hash: {:?}, seed:{}", hash. vm.seed);
```

Comment (philipr-za) [reply base_layer/core/src/proof_of_work/monero_rx/helpers.rs:54]: That is contained in the log line above this already that prints all the monero PoW data.

File: applications/tari_merge_mining_proxy/src/proxy.rs (hunk 1 above)
Comment (philipr-za) [reply applications/tari_merge_mining_proxy/src/proxy.rs:279]: I believe this in the monero_data:
```
DEBUG Valid Monero data: MoneroPowData { header: BlockHeader { major_version: 14, minor_version: 14, timestamp: 1630388505, prev_id: 0xa1d5ac08de1ad6d71f2fa693d0a0182a56b71a073897de57d900cd31aa0726e6, nonce: 7946 }, randomx_key: FixedByteArray { elems: [135, 37, 5, 251, 218, 96, 148, 69, 91, 89, 80, 225, 73, 195, 171, 8, 53, 50, 49, 30, 150, 223, 191, 30, 103, 5, 224, 209, 140, 238, 150, 151, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], len: 32 }, transaction_count: 13, merkle_root: 0x4291a3121f9cd70e82796ae55068b53b307c4b1af1f8994bdc965a130f490b02, coinbase_merkle_proof: MerkleProof { branch: [0xe5f319546804c3a2be6bf401882e1fd5c31b5fdd0cfe2a667b5d97d01de61513, 0xb0e403b00c1afed35bfc30cfa1ffcfa83273bcbde4c5fbe3af9f5c65c6509f6d, 0x315eeb582afe9b2c1ab3ca63c0ecd6a73c847507cbe6e63487ff7d0882f2abdd], depth: 3, path_bitmap: 0 }, coinbase_tx: Transaction { prefix: TransactionPrefix { version: 2, unlock_time: 2438984, inputs: [Gen { height: 2438924 }], outputs: [TxOut { amount: 883916870556, target: ToKey { key: 01cd8ac99334469e8d3e0c1398d38a6e9ef7d895ee15fdf423f111439e9f7d2b } }], extra: ExtraField([TxPublicKey(54e85c64c1a0aa0580d22036fb35abb4f1fe09e580790564762250705593a820), Nonce([0, 8, 168, 19, 82, 253, 252, 0]), MergeMining(0, 0x1b75263e40d83c8423ff193aa754ef428718a95bccfd4068f7b06b28b3ddeb27)]) }, signatures: [], rct_signatures: RctSig { sig: Some(RctSigBase { rct_type: Null, txn_fee: 0, pseudo_outs: [], ecdh_info: [], out_pk: [] }), p: None } } }
```

Comment (philipr-za) [reply applications/tari_merge_mining_proxy/src/proxy.rs:279]: As the randomx_key

**Review by stringhandler — APPROVED** [review]

**Pull request merged**